### DISCUSSION TITLE
Is there a decent way to use tailwind 4 alongside tailwind 3 in a project?

### DISCUSSION THREAD

**Discussion by nathangross** [tailwindlabs/tailwindcss#13575] [Help]
I'm working on a new laravel (+ vite) project that I'd like to utilize tailwind 4 on the frontend but keep tailwind 3 on the backend (filament) due to some tailwind plugin requirements.

I've tried npm workspaces (also yarn), aliasing tailwind3, tailwind4 packages. I'll get the frontend working but it then it breaks the backend and vice versa.

Any thoughts?

**Comment by m-ristau** [comment]
I may be late, but if anyone else ends up here, [this](https://filamentthemes.com/guides/how-to-use-filament-with-tailwind-css-v4) was very helpful.

**Comment by nathangross** [comment]
I was able to solve this with pnpm + workspaces. I wrote up a bit about how to set it up. I't specific to tailwind + filament but would work in other similar situations as well.

https://nathangross.me/blog/using-filament-v3-with-tailwind-css-v4-a-pnpm-workspaces-approach/
